### ISSUE TITLE
Cache Python dependencies in CI

### ISSUE THREAD

**Issue by plannigan** [wayfair-incubator/columbo#292]
`actions/setup-python` supports [caching-packages-dependencies](https://github.com/actions/setup-python#caching-packages-dependencies). Add support for caching for both `requirements.txt` and `requirements-test.txt`.

**Comment by Kaos599** [comment]
hey can i try working on this @plannigan. can you make this to a hacktoberfest label ?

**Comment by plannigan** [comment]
Feel free to take a try. I added the label.

**Comment by Kaos599** [comment]
> Feel free to take a try. I added the label.

there is no requirements.txt and requirements-test.txt in this repo right?
except for requirements-bootstrap.txt? so should i create them

**Comment by plannigan** [comment]
All of the dependencies are in `pyproject.toml`. The library dependencies [are here](https://github.com/plannigan/columbo/blob/main/pyproject.toml#L26C1-L29C1). The development dependencies are within the `tool.hatch.envs` sections.

**Comment by Kaos599** [comment]
Everything seems to be working, except the codecov token. So should i put up a PR for you to review? @plannigan

**Comment by plannigan** [comment]
Yes. Once you think things are in a good state, submit the PR and I'll take a look.

**Comment by pranshu-saraswat** [comment]
@plannigan  Is this issue still open for contribution?
